### DISCUSSION TITLE
How to test on custom images at run time?

### DISCUSSION THREAD

**Discussion by ashmalvayani** [roboflow/notebooks#74] [Q&A] [answered]
### Search before asking

- [X] I have searched the Roboflow Notebooks [issues](https://github.com/roboflow-ai/notebooks/issues) and found no similar bug report.

### Description

The evaluation code block in your notebook reads the valid_ds and use iterators to extract inputs and labels to predict the classes further. I want to test my model on run time for which I do not have any labels. For eg: I have 5 images of digits 1, 2, 3, 4, 5 respectively (I know labels because I can see them, no written csv label files).

How will I do that?

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by github-actions** [comment]
👋 Hello @ashmalvayani, thank you for leaving an issue on Roboflow Notebooks.

## 🐞 Bug reports

If you are filing a bug report, please be as detailed as possible. This will help us more easily diagnose and resolve the problem you are facing. To learn more about contributing, check out our [Contributing Guidelines](https://github.com/roboflow/notebooks/blob/main/CONTRIBUTING.md).

If you require support with custom code that is not part of Roboflow Notebooks, please reach out on the [Roboflow Forum](https://discuss.roboflow.com) or on the [GitHub Discussions page](https://github.com/roboflow/notebooks/discussions) associated with this repository.

## 💬 Get in touch

Do you have more questions about Roboflow that we haven't responded to yet? Feel free to ask them on the [Roboflow Discuss](https://discuss.roboflow.com) forum. Our developer advocates and community team actively respond to questions there.

To ask questions about Notebooks, head over to the [GitHub Discussions](https://github.com/roboflow/notebooks/discussions) section of this repository.

**Comment by SkalskiP** [comment]
Hi, @ashmalvayani 👋🏻! LEt me first move that conversation to our QA section.

**Comment by SkalskiP** [comment]
I'm not sure which notebook we are talking about, but in general:
- If you have ground truth labels, you use `test` to automatically compare model results with labels.
- If you don't have ground truth labels, you use `predict` to save detections locally and evaluate results manually.

**Reply by ashmalvayani** [reply]
Will you be able to show a snippet for how to test a single image without converting it into a test dataloader?

**Reply by SkalskiP** [reply]
Once again, could you tell me what notebook we are talking about?

**Reply by ashmalvayani** [reply]
train-vision-transformer-classification-on-custom-data.ipynb

**Comment by ashmalvayani** [comment, marked as answer]
Here's the code for the problem I asked. SOLVED.

#working perfectly for a single image testing
image_path = '/content/1.PNG'
img = cv2.imread(image_path)

h,w,c = img.shape
img = np.transpose(img, (2, 0, 1))
img = np.expand_dims(img,axis=0)

with torch.no_grad():
  inputs = torch.from_numpy(img)
  inputs = inputs[0].permute(1, 2, 0)

  originalInput = inputs
  for index, array in enumerate(inputs):
    inputs[index] = np.squeeze(array)
  
  inputs = torch.tensor(np.stack(feature_extractor(inputs)['pixel_values'], axis=0))
  inputs = inputs.to(device)

  prediction, loss = model(inputs)

  predicted_class = np.argmax(prediction.cpu())
  plt.imshow(originalInput)
  plt.xlim(224,0)
  plt.ylim(224,0)
  plt.title(f'Prediction: {predicted_class}')
  plt.show()
